src/usr/lib/rsetup/tui/system/system.sh

```diff
@@ -151,7 +151,47 @@ You still need a valid bootloader on another location for the system to work.
 Are you sure you want to erase the bootloader?"
 }
 
-__system() {
+__system_set_target(){
+    radiolist_init
+
+    local current_target available_targets i new_target
+    current_target="$(systemctl get-default)"
+    available_targets=(multi-user.target graphical.target)
+
+    for i in "${available_targets[@]}"
+    do
+        if [[ "$(systemctl list-units --legend=false "$i")" != "" ]]
```

Comment:
```suggestion
        if [[ -z "$(systemctl list-units --legend=false "$i")" ]]
```
我们其他地方的空/非空判断应该都是用-z/-n来完成的